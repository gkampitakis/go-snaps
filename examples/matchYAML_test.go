package examples

import (
	"os"
	"testing"

	"github.com/gkampitakis/go-snaps/match"
	"github.com/gkampitakis/go-snaps/snaps"
)

func TestMatchYaml(t *testing.T) {
	t.Run("should match string yaml", func(t *testing.T) {
		const doc_yaml = `
ser: "mock-service"
name: John Doe
age: 30
email: john.doe@example.com
address: "123 Main St"
street: 123 Main St`

		snaps.WithConfig(snaps.Update(os.Getenv("UPDATE_ONLY") == "true")).
			// .address panics
			MatchYAML(t, doc_yaml, match.Any("$.address", "$.street").Placeholder("mock-address"))
	})

	t.Run("should match struct yaml", func(t *testing.T) {
		type User struct {
			Service string `yaml:"ser"`
			Name    string `yaml:"name"`
			Age     int    `yaml:"age"`
			Email   string `yaml:"email"`
			Address string `yaml:"address"`
			Street  string `yaml:"street"`
		}

		snaps.WithConfig(snaps.Update(os.Getenv("UPDATE_ONLY") == "true")).
			MatchYAML(t, User{
				Service: "mock-service",
				Name:    "John Doe",
				Age:     30,
				Email:   "john.doe@example.com",
				Address: "123 Main St",
				Street:  "123 Main St",
			}, match.Any("$.address", "$.street").Placeholder("mock-address"))
	})
}
