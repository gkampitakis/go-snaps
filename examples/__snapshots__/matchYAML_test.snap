
[TestMatchYaml/should_match_struct_yaml - 1]
name: John Doe
age: 30
email: john.doe@example.com
address: mock-address
time: mock-time

---

[TestMatchYaml/custom_matching_logic - 1]
name: mock-user
email: mock-user@email.com
keys:
  - 1
  - 2
  - 3
  - 4
  - 5

---

[TestMatchYaml/type_matcher - 1]
data: <Type:uint64>
---

[TestMatchYaml/type_matcher - 2]
metadata: <Type:map[string]interface {}>
---
